### PR TITLE
Remove return value from opaque build function

### LINKED ISSUE #1683: Multiple opaque processing functions should not return status
For example : https://github.com/DMTF/libspdm/blob/4853e80f8aa14f4893941f1986f758353541be1d/library/spdm_requester_lib/libspdm_req_common.c#L48-L51

https://github.com/DMTF/libspdm/blob/4853e80f8aa14f4893941f1986f758353541be1d/library/spdm_requester_lib/libspdm_req_key_exchange.c#L386-L388

### PR DESCRIPTION
**Pull request by steven-bellock**
Fixes #1683.

Signed-off-by: Steven Bellock <sbellock@nvidia.com>

### CODE REVIEW

File: library/spdm_requester_lib/libspdm_req_common.c [hunk 1]
```diff
@@ -24,11 +24,11 @@ uint16_t libspdm_allocate_req_session_id(libspdm_context_t *spdm_context)
...
 
-libspdm_return_t libspdm_build_opaque_data_supported_version_data(libspdm_context_t *spdm_context,
-                                                                  size_t *data_out_size,
-                                                                  void *data_out)
+void libspdm_build_opaque_data_supported_version_data(libspdm_context_t *spdm_context,
+                                                      size_t *data_out_size,
+                                                      void *data_out)
 {
-    size_t final_data_size;
+    size_t final_data_size = libspdm_get_opaque_data_supported_version_data_size(spdm_context);
```
Comment (jyao1) [inline library/spdm_requester_lib/libspdm_req_common.c:31]: I don't recommend we do assignment in definition.

Can we move it just before ASSERT?

```
final_data_size = libspdm_get_opaque_data_supported_version_data_size(spdm_context);
LIBSPDM_ASSERT(*data_out_size >= final_data_size);
```

File: library/spdm_responder_lib/libspdm_rsp_common.c [hunk 2]
```diff
@@ -24,12 +24,11 @@ uint16_t libspdm_allocate_rsp_session_id(const libspdm_context_t *spdm_context)
...
-libspdm_return_t
-libspdm_build_opaque_data_version_selection_data(const libspdm_context_t *spdm_context,
-                                                 size_t *data_out_size,
-                                                 void *data_out)
+void libspdm_build_opaque_data_version_selection_data(const libspdm_context_t *spdm_context,
+                                                      size_t *data_out_size,
+                                                      void *data_out)
 {
-    size_t final_data_size;
+    size_t final_data_size = libspdm_get_opaque_data_version_selection_data_size(spdm_context);
```
Comment (jyao1) [inline library/spdm_responder_lib/libspdm_rsp_common.c:31]: I don't recommend we do assignment in definition.

Can we move it just before ASSERT?

File: library/spdm_requester_lib/libspdm_req_common.c (hunk 1 above)
Comment (steven-bellock) [reply library/spdm_requester_lib/libspdm_req_common.c:31]: Will do.

File: library/spdm_responder_lib/libspdm_rsp_common.c (hunk 2 above)
Comment (steven-bellock) [reply library/spdm_responder_lib/libspdm_rsp_common.c:31]: Will do.

**Pull request merged**